SteamKit2/SteamKit2/Networking/Steam3/TcpConnection.cs

```diff
@@ -152,12 +152,24 @@ private void TryConnect(object sender)
             int timeout = (int)sender;
             if (cancellationToken.IsCancellationRequested)
             {
-                log.LogDebug( "TcpConnection", "Connection to {0} cancelled by user", CurrentEndPoint);
+                log.LogDebug( nameof( TcpConnection ), "Connection to {0} cancelled by user", CurrentEndPoint );
                 Release( userRequestedDisconnect: true );
                 return;
             }
-            
-            var asyncResult = socket!.BeginConnect(CurrentEndPoint, null, null );
+
+            IAsyncResult asyncResult;
+
+            try
+            {
+                asyncResult = socket!.BeginConnect( CurrentEndPoint, null, null );
+            }
+            catch ( Exception ex )
+            {
+                log.LogDebug( nameof( TcpConnection ), "Socket exception while beginning connection request to {0}: {1}", CurrentEndPoint, ex );

```

Comment:
We're catching any old `Exception` here, so it might not necessarily be a `"Socket exception..."`.